core/src/test/java/io/questdb/griffin/DropIndexTest.java

```diff
@@ -0,0 +1,182 @@
+        assertMemoryLeak(() -> {
+            compiler.compile(
+                    "create table sensors as (\n" +
+                            "    select \n" +
+                            "        rnd_symbol('ALPHA', 'OMEGA', 'THETA') sensor_id, \n" +
+                            "        rnd_int() temperature, \n" +
+                            "        timestamp_sequence(0, 36000000) ts \n" +
+                            "    from long_sequence(100)\n" +
+                            ") timestamp(ts) partition by DAY",
+                    sqlExecutionContext
+            );
+
+            try {
+                compile("alter table sensors alter column sensor_id drop index", sqlExecutionContext);
+                Assert.fail();
+            } catch (SqlException e) {
+                Assert.assertEquals(12, e.getPosition());
+                TestUtils.assertContains(e.getFlyweightMessage(), "column 'sensor_id' is not indexed");
+            }
+        });
+    }
+
+    @Test
+    public void testDropIndexOfIndexedColumnPartitionedTable() throws Exception {
+        assertMemoryLeak(() -> {
+            compiler.compile(
+                    "create table sensors as (\n" +
+                            "    select\n" +
+                            "        rnd_symbol('ALPHA', 'OMEGA', 'THETA') sensor_id, \n" +
+                            "        rnd_int() temperature, \n" +
+                            "        timestamp_sequence(0, 36000000) ts \n" +
+                            "    from long_sequence(1000)\n" +
+                            "), index(sensor_id capacity 32) timestamp(ts) partition by HOUR",
+                    sqlExecutionContext
+            );
+            verifyIndexMetadata("sensors", "sensor_id", true, 32);
+
+            compile("alter table sensors alter column sensor_id drop index", sqlExecutionContext);
+            verifyIndexMetadata("sensors", "sensor_id", false, configuration.getIndexValueBlockSize());
+        });
+    }
+
+    @Test
+    public void testDropIndexOfIndexedColumnNonPartitionedTable() throws Exception {
+        assertMemoryLeak(() -> {
+            compiler.compile(
+                    "create table sensors as (\n" +
+                            "    select\n" +
+                            "        rnd_symbol('ALPHA', 'OMEGA', 'THETA') sensor_id, \n" +
+                            "        rnd_int() temperature, \n" +
+                            "        timestamp_sequence(0, 36000000) ts \n" +
+                            "    from long_sequence(100)\n" +
+                            "), index(sensor_id capacity 32) timestamp(ts) partition by NONE",
+                    sqlExecutionContext
+            );
+            verifyIndexMetadata("sensors", "sensor_id", true, 32);
+
+            compile("alter table sensors alter column sensor_id drop index", sqlExecutionContext);
+            verifyIndexMetadata("sensors", "sensor_id", false, configuration.getIndexValueBlockSize());
```

Comment:
we would also need to check that already existing TableReader instance can see that metadata changed after reload() call